modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/Utils.java

```diff
@@ -208,6 +212,51 @@
         MOSAIC_SUPPORT_FILES_FILTER = initMosaicSupportFilesFilter();
     }
 
+    /** Check if the provided reader is a MultiCRS Reader and it can support the specified crs. */
+    public static boolean isSupportedCRS(GridCoverage2DReader reader, CoordinateReferenceSystem crs)
+            throws FactoryException, IOException {
+        String multiCrsReader =
+                reader.getMetadataValue(AbstractGridCoverage2DReader.MULTICRS_READER);
+        boolean isMultiCRS = multiCrsReader != null && Boolean.valueOf(multiCrsReader);
+        String supportedEpsgCodes = "";
+        if (isMultiCRS) {
+            supportedEpsgCodes =
+                    reader.getMetadataValue(AbstractGridCoverage2DReader.MULTICRS_EPSGCODES);
+        }
+        if (!StringUtils.isBlank(supportedEpsgCodes) && crs != null) {
+            Integer targetCRSEpsgCode = CRS.lookupEpsgCode(crs, true);
+            if (targetCRSEpsgCode != null) {
+                String code = String.valueOf(targetCRSEpsgCode);
+                supportedEpsgCodes = supportedEpsgCodes.replaceAll("[^0-9,]", "");
+                String[] epsgCodes = supportedEpsgCodes.split(",");
+                for (String epsgCode : epsgCodes) {
+                    if (code.equalsIgnoreCase(epsgCode)) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Return the EPSG code number of a provided coordinate reference system by just returning the
+     * identifier's code (not doing any lookup or scan).
+     *
+     * @param crs the CoordinateReferenceSystem to be checked
+     * @return null if no code is available
+     */
+    public static String getEpsgCodeNumber(CoordinateReferenceSystem crs) {
```

Comment:
This looks like a broken version of "CRS.lookupEpsgCode(crs, false)", in that it will fail if the CRS happens to have alternative authorities that are not "EPSG". Setting the second parameter to "false" avoids scans.

Reply:
Thanks for the notice. I will fix it.